diff --git a/ini/GLideN64.custom.ini b/ini/GLideN64.custom.ini
index ba75893d..7fe7c58a 100644
--- a/ini/GLideN64.custom.ini
+++ b/ini/GLideN64.custom.ini
@@ -78,7 +78,7 @@ frameBufferEmulation\copyAuxToRDRAM=1
 
 [DONKEY%20KONG%2064]
 Good_Name=Donkey Kong 64 (E)(J)(U)
-frameBufferEmulation\copyDepthToRDRAM=1
+frameBufferEmulation\copyDepthToRDRAM=2
 frameBufferEmulation\N64DepthCompare=0
 
 [DR.MARIO%2064]
@@ -361,7 +361,7 @@ frameBufferEmulation\copyToRDRAM=1
 
 [TUROK_DINOSAUR_HUNTE]
 Good_Name=Turok - Dinosaur Hunter (E)(G)(U)(J)
-frameBufferEmulation\copyDepthToRDRAM=1
+frameBufferEmulation\copyDepthToRDRAM=2
 
 [WAVE%20RACE%2064]
 Good_Name=Wave Race 64 (E)(U)(J)
