diff --git a/yabause/CMake/Packages/external_libchdr.cmake b/yabause/CMake/Packages/external_libchdr.cmake
index f6a8db60..0bd9958a 100644
--- a/yabause/CMake/Packages/external_libchdr.cmake
+++ b/yabause/CMake/Packages/external_libchdr.cmake
@@ -23,6 +23,13 @@ elseif(IOS)
     -DCMAKE_TOOLCHAIN_FILE=${TOOL_CHAIN_ABSOLUTE_PATH}
   )
 
+elseif(CMAKE_CROSSCOMPILING AND CMAKE_TOOLCHAIN_FILE)
+  get_filename_component(TOOL_CHAIN_ABSOLUTE_PATH "${CMAKE_TOOLCHAIN_FILE}"
+                         REALPATH BASE_DIR "${CMAKE_BINARY_DIR}")
+  set(ADDITIONAL_CMAKE_ARGS
+    -DCMAKE_TOOLCHAIN_FILE=${TOOL_CHAIN_ABSOLUTE_PATH}
+    -DWITH_LZMA_ASM=OFF
+  )
 else()
   set(ADDITIONAL_CMAKE_ARGS "")
 endif()
